clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -4951,6 +4960,10 @@ void CXXNameMangler::mangleExpression(const Expr *E, unsigned Arity,
     mangleFixedPointLiteral();
     break;
 
+  case Expr::DecimalFloatLiteralClass:
+    mangleType(E->getType());
```

Comment:
The literal value will need to be mangled here as well; not just the type. See https://godbolt.org/z/Yd8a9ofGn for an example (and a chance to chuckle at the icc behavior).